proposals/p1964.md

```diff
@@ -0,0 +1,351 @@
+    -   [No Distinct Character Literal](#no-distinct-character-literal)
+    -   [Supporting Prefix Declarations](#supporting-prefix-declarations)
+    -   [Allowing Numeric Escape Sequences](#allowing-numeric-escape-sequences)
+
+<!-- tocstop -->
+
+## Abstract
+
+This proposal specifies lexical rules for constant characters in Carbon:
+
+Put character literals in single quotes, like `'a'`. Character literals work
+like numeric literals:
+
+-   Every different literal value has its own type.
+-   The bit width is determined by the type of the variable the literal is
+    assigned to, not the literal itself.
+-   A character literal must contain exactly one code point.
+
+Follows the plan from open design idea
+[#1934: Character Literals](https://github.com/carbon-language/carbon-lang/issues/1934).
+
+## Problem
+
+Carbon currently has no lexical syntax for character literals, and only provides
+string literals and numeric literals. We wish to provide a distinct lexical
+syntax for character literals versus string literals.
+
+The advantage of having an explicit character type fundamentally comes down to
+characters being represented as integers whereas strings are represented as
+buffers. This will allow characters to have different operations, and be more
+familiar to use. For example:
+
+```
+if (c >= 'A' and c <= 'Z') {
+    c += 'a' - 'A';
+}
+```
+
+The example above shows how we would be able to use operations similar to
+integers. Being able to use the comparison operations and supporting arithmetic
+operations provides an intuitive approach to using characters. This allows us to
+remove unnecessary logic of type conversion and other control flow logic, that
+is needed to work with a single element string. See [Rationale](#rationale) for
+more examples showing more appropriate use of characters over using strings.
+
+## Background
+
+Character Literals by definition is a type of literal in programming for the
+representation of a single character's value within the source code of a
+computer program. Character literals between languages have some minor nuances
+but are fundamentally designed for the same purpose. Languages that have a
+dedicated character data type generally include character literals, for example
+C++, Java, Swift to name a few. Whereas other languages that lack distinct
+character type, like Python use strings of length one to serve the same purpose
+a character data type. For more information see
+[Character Literals Wiki](https://en.wikipedia.org/wiki/Character_literal),
+[Character Literals DBpedia](https://dbpedia.org/page/Character_literal)
+
+## Proposal
+
+Put character literals in single quotes, like `'a'`. Character literals work
+like numeric literals:
+
+-   Every different literal value has its own type.
+-   The bit width is determined by the type of the variable the literal is
+    assigned to, not the literal itself. Follows the plan from #1934.
```

Comment:
I would adjust this to state up-front that the core of the design is to model code points and the fact that they have a single concrete numerically representable value.

I think you should also clarify at the beginning that this means *not* supporting the full space of grapheme clusters or direct modeling of specific code *unit* sequences, which should instead be modeled with normal string literals.